### ISSUE TITLE
OpenID AUTH returns `invalid credentials`

### ISSUE THREAD

**Issue by ithan** [directus/directus#19003]
### Describe the Bug

[x] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
[x] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
[~] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.

There are similar closed issues, but none seems to be able to help for this.

I'm trying to use [Go Authentik](https://goauthentik.io/) with OpenID as the primary oauth system for Directus.

My problem is that after successfully login in from Authentik, Directus will return invalid credentials

```
    err: {

      "type": "InvalidCredentialsException",

      "message": "Invalid user credentials.",

      "stack":

          Error: Invalid user credentials.
           ...
      "status": 401,

      "code": "INVALID_CREDENTIALS",

      "extensions": {}
```

In the env, I have "AUTH_AUTHENTIK_IDENTIFIER_KEY": "email", and the email from directus and authentik ( for the external identifier ) match.

Authentik response successfully provided the `{ pk: number, email: string, username: string }` response.

### To Reproduce

Try to access directus using OpenID ( Only tested with GoAuthentik )

### Directus Version

v10.3.0

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rijkvanzanten** [comment]
Could you share the full set of environment variables? Are there any logs available where you're running Directus itself? It should log what caused the invalid credentials error to be thrown 👍🏻

**Comment by ithan** [comment]
Hello! Absolutely!

Here are the env variables


```json
"AUTH_PROVIDERS": "authentik"
"AUTH_AUTHENTIK_ISSUER_URL": "https://{url}/application/o/directus/.well-known/openid-configuration"
"AUTH_AUTHENTIK_DRIVER": "openid"
"AUTH_AUTHENTIK_IDENTIFIER_KEY": "email"
"AUTH_AUTHENTIK_CLIENT_ID": "***"
"AUTH_AUTHENTIK_CLIENT_SECRET": "***"
/** and the rest of the required variables **/
```


The error:
```
[06:06:43] GET /auth/login/authentik/callback?code=*** 302 513ms

[06:06:43] GET /admin/login?reason=INVALID_CREDENTIALS 200 1ms

[06:06:44] GET /admin/assets/index.aa47f040.entry.js 200 15ms

[06:06:43.940] WARN: Invalid user credentials.

    err: {

      "type": "InvalidCredentialsException",

      "message": "Invalid user credentials.",

      "stack":

          Error: Invalid user credentials.

              at handleError (file:///directus/node_modules/.pnpm/file+api/node_modules/@directus/api/dist/auth/drivers/openid.js:220:16)

              at OpenIDAuthDriver.getUserID (file:///directus/node_modules/.pnpm/file+api/node_modules/@directus/api/dist/auth/drivers/openid.js:114:19)

              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

              at async AuthenticationService.login (file:///directus/node_modules/.pnpm/file+api/node_modules/@directus/api/dist/services/authentication.js:42:22)

              at async file:///directus/node_modules/.pnpm/file+api/node_modules/@directus/api/dist/auth/drivers/openid.js:273:28

      "status": 401,

      "code": "INVALID_CREDENTIALS",

      "extensions": {}

    }
```

Authentik logs show that the authentification was successful, and the email in directus & authentik match

**Comment by rijkvanzanten** [comment]
After an hour or so of messing around with the Authentik containers, I was able to get it running locally and was able to register users and use said users from Directus without a problem 🤔 

I think fetching the user information is failing against your specific configuration in Authentik in this bit:

https://github.com/directus/directus/blob/6984395bdbf9badb676314563180ccec79864d44/api/src/auth/drivers/openid.ts#L138-L158

The shared stack trace makes me think it's that `handleError` call there that's calling it, which then converts the unknown third party error into Directus' `InvalidCredentialsError` here:

https://github.com/directus/directus/blob/6984395bdbf9badb676314563180ccec79864d44/api/src/auth/drivers/openid.ts#L287-L292

Could set Directus' `LOG_LEVEL` to `trace` and check / share the error that's then logged? Thanks!

**Comment by ithan** [comment]
Well, I found the problem! And it was not really caused by Directus, but rather a misconfiguration on my end.

The error that I got was 
```
TRACE: [OpenID] Unknown RP error
    err: {
      "type": "RPError",
      "message": "unexpected JWT alg received, expected RS256, got: HS256",
      "stack":
          RPError: unexpected JWT alg received, expected RS256, got: HS256
```

And googling a bit, I found this https://github.com/goauthentik/authentik/issues/3909

It wasn't the same issue, but this give me the clue, "Signing Key" was not properly selected for the provider, selecting a self signed certificate solved the issue! 

Thank you for your inside! Hopefully this can serve as a reference if someone has the same problem!
